### PR TITLE
blog first draft

### PR DESCRIPTION
**Pull request by wbentley15**
Bios:
Walter Bentley – Rackspace Private Cloud Solutions Architect – Walter is a new Racker with a diverse background in Production Systems Administration and Solutions Architecture. He brings over 15 years of experience across numerous industries such as Online Marketing, Financial, Insurance, Aviation, Food Industry and Education. In the past, always being the requester, consumer and advisor to companies to use technologies such as OpenStack, now promoter of OpenStack technology and Cloud educator.

Twitter: @djstayflypro
LinkedIn: http://goo.gl/r2p21i
Github: wbentley15

Accept to allow redaction, editing or structure changes by technical writer staff desired date of publishing
Publish date: 07-21-2015

### CODE REVIEW

**Comment by wbentley15** [comment]
@catlook please do your magic, it is a short blog this time around :+1:

File: _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+author: Walter Bentley
+published: true
+categories:
+    - rackspace-private-cloud
+    - OSAD
+    - OpenStack
+    - Ansible
+---
+
+So you have spent months convincing your leadership to go with OpenStack.  Finally the keys of the cloud are turned over to you as the Cloud Operator, you then look over at your co-workers and say “now what”.  The next set of phrases normal are something like: Now how do we best administer this cloud?  Cloud is suppose to be easier, right?
```
Comment (catlook) [inline _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md]: ... Cloud Operator, and you then look over ...
... phrases normally are ...

File: _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+<!-- more -->
+
+Well do not worry, cloud can be easier and better depending on how you approach it.  As OpenStack has begun to be considered more of a mainstream cloud platform, the challenge of operating it after it is built has become prevalent.  While all cloud tasks can be executed via the API or CLI tool on a one by one basis, this would not be the best way to handle larger cloud deployments.  The need for more of an automated approach to administering OpenStack is now clear.  With the many IaaS capabilities OpenStack has to offer coupled with Ansible, an ease of use configuration management tool, assures a more complete cloud implementation.
```
Comment (catlook) [inline _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md]: ... not worry! Cloud can be ...
... need for a more automated approach ...
... is now clear.  The many IaaS capabilities OpenStack has to offer coupled with Ansible, an easy-to-use configuration management tool, provide a more complete cloud implementation.

File: _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* Ansible playbooks can be written against the OpenStack API’s or Python CLI’s
+* Designing roles with unique variable values within playbooks/roles is as easy as writing a email
+* Pre-built structure to create dynamic inventory scripts within Ansible
+
+Below is a quick visual of how the flow of using Ansible to handle various administrative tasks within your OpenStack cloud.  In this flow starting from the top, you as the Cloud Operator would consume Ansible playbooks that were written by the DevOps team (typically this would be the administrators working with the developers, making up that DevOps dynamic).  From there the playbooks can be executed against the OpenStack API’s directly and/or using the services CLI’s.  Personally I prefer to use the OpenStack CLI’s at this point as the Ansible OpenStack modules are not as mature as I prefer at this current time.  This issue is being actively addressed, so much so a half day session was dedicated to this topic at the last OpenStack Summit hosted in Vancouver.  Finally at this point the Cloud Consumer, depicted on the right in the flow, can utilize the resources created and/or capabilities add with the administration playbooks.
```
Comment (catlook) [inline _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md]: ... From there, the playbooks ...
... API’s directly or by using the ...
... Personally, I prefer ...
... actively addressed.  In fact, a half day ...
... Finally, at this point,  the Cloud ...
... resources created by, or the capabilities added with, the administration ...

File: _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+<pic>
+
+With the above approach in mind, you can go off and begin creating serious of playbooks/roles to handle typical daily Cloud Operator type tasks.  During the webinar I presented last week, we walked thru some possible scenarios and how they could be easily/swiftly solved for using Ansible.  Instead of stepping thru them again, I felt it would be best to just provide the location on GitHub where you could review them and/or give them a try on your own.  Those working examples can be found at:  [https://github.com/wbentley15/ansiblefest-demo](https://github.com/wbentley15/ansiblefest-demo).  These examples where the very same demos shown during [AnsibleFest](http://www.ansible.com/blog/ansiblefest-nyc-speakers) held in New York this year.
```
Comment (catlook) [inline _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md]: ... review them and give them  ...
... These examples are the very ...

File: _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+In conclusion, I have found much success in helping customers with their OpenStack clouds by leveraging Ansible.  The Rackspace approach of deploying OpenStack, within our datacenter and/or within a customers datacenters, is to use [OSAD (OpenStack Ansible Deployment)](https://github.com/stackforge/os-ansible-deployment), so taking it to the next level to handle operator tasks seemed very obvious.  As I call it, a two for one capability.  If you are interested in learning more about using Ansible to operate your OpenStack cloud, please be on the look out for my book entitled “OpenStack Administration with Ansible” set to hit online/in-stores end of this year.  Promise it will be a good read and reference when creating your OpenStack playbooks.
```
Comment (catlook) [inline _posts/2015-07-21-operating-your-openstack-cloud-using-ansible.md]: ... datacenter or within a customer's datacenter, is ...  [ ed: or you could say this: ... within customers' datacenters, is ... ]
.. As I call it, it's a two-for-one capability...

**Comment by wbentley15** [comment]
@catlook as always thank you for your help and kind words on the book deal.
@rgbkrk ready to be shipped...all aboard :accept:

**Comment by rgbkrk** [comment]
Yay for @ktbartholomew!

**Comment by rgbkrk** [comment]
Great article @wbentley15!

**Comment by wbentley15** [comment]
Thanks!

Walter

**Pull request merged**